src/System.CommandLine.Tests/ParseDirectiveTests.cs

```diff
@@ -46,6 +46,56 @@ public async Task Parse_directive_writes_parse_diagram()
                    .Be($"![ {RootCommand.ExecutableName} [ subcommand [ -c <34> ] ] ]   ???--> --nonexistent wat" + Environment.NewLine);
         }
 
+        [Fact]
+        public async Task When_parse_directive_is_used_the_help_is_not_displayed()
```

Comment:
we were missing test coverage scenario for cases when users combine  `[parse]` and `--help` or `--version`